### ISSUE TITLE
How to obtain bbox file?

### ISSUE THREAD

**Issue by qiuzhongwei-USTB** [karfly/learnable-triangulation-pytorch#26]
Hi, could you tell me how to generate bounding box file?
Or, could you provide the bounding box files you used?

**Comment by shrubb** [comment]
Hi @qiuzhongwei-USTB,

for Human3.6M, follow the instructions in [here](https://github.com/karfly/learnable-triangulation-pytorch/tree/master/mvn/datasets/human36m_preprocessing) for ground truth boxes, or [see this](https://github.com/karfly/learnable-triangulation-pytorch/issues/19#issuecomment-545993330) Mask R-CNN boxes;

for CMU Panoptic, the bounding boxes are not yet ready, as is the whole pipeline, sorry for that.

**Comment by qiuzhongwei-USTB** [comment]
Thanks，
could you tell me what's the difference of using undistort images and origin image?

**Comment by shrubb** [comment]
Please read about camera models, for example, [here](https://docs.opencv.org/2.4/modules/calib3d/doc/camera_calibration_and_3d_reconstruction.html). With original images, the ground truth 3D points will project to slightly inaccurate locations in images (still may be OK, but the quality will be lower). Undistortion corrects this.

**Comment by qiuzhongwei-USTB** [comment]
Thanks for your reply! I met some problems again. 

1) I found that it's very slow when I evaluate your pre-trained model(about 70 s/img, one V100 GPU).
Could provide your train times and test times on Human3.6 dataset?

2) There is an error  that when I set 'retain_every_n_frames_in_test=5 or 64'.


error information：
Experiment name: eval_human36m_ransac_RANSACTriangulationNet@08.11.2019-13:24:58
5it [06:07, 73.45s/it]
/root/v-helzhe/zw/h36m/data/learnable-triangulation-pytorch/mvn/datasets/human36m.py:222: RuntimeWarning: invalid value encountered in true_divide
  action_scores[k] = v['total_loss'] / v['frame_count']
/root/v-helzhe/zw/h36m/data/learnable-triangulation-pytorch/mvn/datasets/human36m.py:222: RuntimeWarning: invalid value encountered in double_scalars
  action_scores[k] = v['total_loss'] / v['frame_count']

**Comment by shrubb** [comment]
Not sure. Let us look into this this after the weekend.

**Comment by shrubb** [comment]
@qiuzhongwei-USTB did you close the issue because you worked around those problems? If yes, how did you solve them?

**Comment by qiuzhongwei-USTB** [comment]
> @qiuzhongwei-USTB did you close the issue because you worked around those problems? If yes, how did you solve them?

Yes, v['frame_count'] could be zeros. I change the code like the follows:
            for k, v in action_scores.items():
                if v['frame_count'] > 0:
                    action_scores[k] = v['total_loss'] / v['frame_count']
                else:
                    action_scores[k] = 0

**Comment by shrubb** [comment]
How about the long evaluation time?

**Comment by qiuzhongwei-USTB** [comment]
I didn't solve these. I can bear the evaluation time when I use 8 GPU with distributed testing.
Can you provide your evaluation time?

**Comment by karfly** [comment]
Hi @qiuzhongwei-USTB!
It's very strange that you get such big evaluation times. I've just checked our evaluation times on full Human3.6M validation dataset:
- Algebraic: 1 hour 29 minutes
- Volumetric: 2 hours 1 minute

We use single Nvidia Tesla P40 GPU for evaluation.

Could you provide full bash command, which you used for evaluation?
